tests/integ/datasource/test_mysql.py

```diff
@@ -0,0 +1,186 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+import logging
+import math
+from decimal import Decimal
+
+import pytest
+
+from snowflake.snowpark import Row
+from snowflake.snowpark._internal.data_source.drivers.pymsql_driver import PymysqlDriver
+from snowflake.snowpark._internal.data_source.utils import DBMS_TYPE
+from tests.resources.test_data_source_dir.test_mysql_data import (
+    mysql_real_data,
+    MysqlType,
+)
+from tests.utils import RUNNING_ON_JENKINS, Utils
+from tests.parameters import MYSQL_CONNECTION_PARAMETERS
+
+DEPENDENCIES_PACKAGE_UNAVAILABLE = True
+try:
+    import databricks  # noqa: F401
+    import pandas  # noqa: F401
+
+    DEPENDENCIES_PACKAGE_UNAVAILABLE = False
+except ImportError:
+    pass
+
+
+pytestmark = [
+    pytest.mark.skipif(DEPENDENCIES_PACKAGE_UNAVAILABLE, reason="Missing 'databricks'"),
+    pytest.mark.skipif(
+        RUNNING_ON_JENKINS, reason="cannot access external datasource from jenkins"
+    ),
+    pytest.mark.skipif(
+        "config.getoption('local_testing_mode', default=False)",
+        reason="feature not available in local testing",
+    ),
+]
+
+TEST_TABLE_NAME = "ALL_TYPES_TABLE"
+MYSQL_TEST_EXTERNAL_ACCESS_INTEGRATION = "mysql_integration"
+
+
+def create_connection_mysql():
+    import pymysql
+
+    conn = pymysql.connect(
+        user=MYSQL_CONNECTION_PARAMETERS["username"],
+        password=MYSQL_CONNECTION_PARAMETERS["password"],
+        host=MYSQL_CONNECTION_PARAMETERS["host"],
+        database=MYSQL_CONNECTION_PARAMETERS["database"],
```

Comment:
nit: you can do `pymysql.connect(**MYSQL_CONNECTION_PARAMETERS)`

Reply:
cannot do it for now as the key value in connection params is a bit different from the arg name in pymysql.connect, will log a jira to fix it